### PR TITLE
Simplify profile reloading

### PR DESCRIPTION
**Pull request by dhaavi**
Also, increase prompt decision timeout.

### CODE REVIEW

File: profile/active.go [hunk 1]
```diff
@@ -58,6 +58,13 @@ func addActiveProfile(profile *Profile) {
 	activeProfilesLock.Lock()
 	defer activeProfilesLock.Unlock()
 
+	// Mark any previous profile as outdated.
+	previous, ok := activeProfiles[profile.ScopedID()]
+	if ok {
+		previous.outdated.Set()
+	}
```
Comment (ppacher) [inline profile/active.go:65]: `previous` and `ok` are not required in the outer scope so consider:
```suggestion
if previous, ok := activeProfiles[profile.ScopedID()]; ok {
    previous.outdated.Set()
}
```

File: profile/get.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ const (
 
 var getProfileSingleInflight singleflight.Group
 
-// GetProfile fetches a profile. This function ensure that the profile loaded
+// GetProfile fetches a profile. This function ensures that the loaded profile
 // is shared among all callers. You must always supply both the scopedID and
 // linkedPath parameters whenever available.
 func GetProfile(source profileSource, id, linkedPath string) ( //nolint:gocognit
```
Comment (ppacher) [inline profile/get.go:34]: IMHO the function documentation does not explain what happes if I don't have the `scopeID` or `linkedPath available`.
We may also rename the type of the first parameter and remove the `profile` prefix as this is the package name and stutters when (trying) to use from somewhere else: `profile.profileSource`

File: profile/get.go [hunk 3]
```diff
@@ -266,13 +192,13 @@ func prepProfile(r record.Record) (*Profile, error) {
...
-		log.Warningf("profiles: profile %s has (partly) invalid configuration: %s", profile.ID, err)
+		log.Errorf("profiles: profile %s has (partly) invalid configuration: %s", profile.ID, err)
 	}
 
 	// parse config
 	err = profile.parseConfig()
 	if err != nil {
-		log.Warningf("profiles: profile %s has (partly) invalid configuration: %s", profile.ID, err)
+		log.Errorf("profiles: profile %s has (partly) invalid configuration: %s", profile.ID, err)
 	}
```
Comment (ppacher) [inline profile/get.go:202]: If that is an error I guess we should `return nil, err`?

File: profile/profile.go [hunk 4]
```diff
@@ -301,18 +298,6 @@ func (profile *Profile) addEndpointyEntry(cfgKey, newEntry string) {
...
-		if !changed || profile.layeredProfile == nil {
-			return
-		}
-
-		profile.layeredProfile.Lock()
-		defer profile.layeredProfile.Unlock()
-
-		profile.layeredProfile.RevisionCounter++
-	}()
-
```
Comment (ppacher) [inline profile/profile.go:315]: How does the revision counter get handled now?

Comment (dhaavi) [reply profile/profile.go:315]: Profiles are marked as outdated when they are saved. The next operation that uses the profile checks for outdated profiles, reloads them and increases the counter. Only having a single way of doing that makes this simpler and less error prone.

File: profile/get.go (hunk 2 above)
Comment (dhaavi) [reply profile/get.go:34]: I improved the documentation.
Yes, `profile.profileSource` stutters a bit, but just using `source` is not explicit enough, IMHO - I tried it.

File: profile/get.go (hunk 3 above)
Comment (dhaavi) [reply profile/get.go:202]: Let's talk about that.

**Pull request merged**